### ISSUE TITLE
Filter not found exception

### ISSUE THREAD

**Issue by rakanalh** [raiden-network/raiden#1823]
## Problem Definition

After running the raiden node for a while, the node crashed because of the following error:
```
2018-07-15 20:21:53 [error    ] alarm missed 1 blocks          [raiden.tasks] current_block=3644548
2018-07-15 20:21:53 [debug    ] new block                      [raiden.tasks] number=3644548
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/rakan/raiden/raiden/tasks.py", line 56, in _run
    self.poll_for_new_block()
  File "/home/rakan/raiden/raiden/tasks.py", line 95, in poll_for_new_block
    result = callback(current_block, chain_id)
  File "/home/rakan/raiden/raiden/raiden_service.py", line 376, in _callback_new_block
    for event in self.blockchain_events.poll_blockchain_events():
  File "/home/rakan/raiden/raiden/blockchain/events.py", line 326, in poll_blockchain_events
    for log_event in getattr(event_listener.filter, query_fn)():
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/web3-4.4.1-py3.6.egg/web3/utils/filters.py", line 105, in get_new_entries
    log_entries = self._filter_valid_entries(self.web3.eth.getFilterChanges(self.filter_id))
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/web3-4.4.1-py3.6.egg/web3/eth.py", line 344, in getFilterChanges
    "eth_getFilterChanges", [filter_id],
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/web3-4.4.1-py3.6.egg/web3/manager.py", line 110, in request_blocking
    raise ValueError(response["error"])
ValueError: {'code': -32000, 'message': 'filter not found'}
2018-07-15T20:21:53Z <AlarmTask "Greenlet-0" at 0x7f150e34c048: _run> failed with ValueError

Traceback (most recent call last):
  File "/home/rakan/.pyenv/versions/raiden/bin/raiden", line 11, in <module>
    load_entry_point('raiden', 'console_scripts', 'raiden')()
  File "/home/rakan/raiden/raiden/__main__.py", line 11, in main
    run(auto_envvar_prefix='RAIDEN')
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/rakan/.pyenv/versions/3.6.5/envs/raiden/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/rakan/raiden/raiden/ui/cli.py", line 966, in run
    app_ = _run_app()
  File "/home/rakan/raiden/raiden/ui/cli.py", line 940, in _run_app
    event.wait()
  File "src/gevent/event.py", line 240, in gevent._event.Event.wait
  File "src/gevent/event.py", line 140, in gevent._event._AbstractLinkable._wait
  File "src/gevent/event.py", line 117, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/event.py", line 119, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 63, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/__greenlet_primitives.pxd", line 35, in gevent.__greenlet_primitives._greenlet_switch
ValueError: {'code': -32000, 'message': 'filter not found'}
```
### System Description
```
 uname -a
Linux archlinux 4.17.4-1-ARCH #1 SMP PREEMPT Tue Jul 3 15:45:09 UTC 2018 x86_64 GNU/Linux
```
```
 command -v solc && solc --version
/usr/bin/solc
solc, the solidity compiler commandline interface
Version: 0.4.24+commit.e67f0147.Linux.g++
```
```
command -v geth && geth version
/usr/bin/geth
Geth
Version: 1.8.11-stable
Git Commit: dea1ce052a10cd7d401a5c04f83f371a06fe293c
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.3
Operating System: linux
GOPATH=/home/rakan/Code/Go
GOROOT=/usr/lib/go
```
```
command -v raiden && raiden version
/home/rakan/.pyenv/shims/raiden
{
  "raiden": "0.3.1.dev984+ga662e898",
  "python_implementation": "CPython",
  "python_version": "3.6.5",
  "system": "Linux 64bit_ELF 4.17.4-1-ARCH x86_64"
}
```
```
[ -d .git ] && git rev-parse HEAD
d7742896843eb11d4c6532e63ca176f1e1c73d11
```

**Comment by LefterisJP** [comment]
And what is the etherreum client you are connecting to?

**Comment by andrevmatos** [comment]
Exactly the error I'm facing because of https://github.com/paritytech/parity/issues/9122

**Comment by LefterisJP** [comment]
We should also recommend geth for this first version of raiden as we have not tested enough with parity and from what we see so far there are quite a few bugs.

**Comment by rakanalh** [comment]
Geth:
```
Geth
Version: 1.8.11-stable
Git Commit: dea1ce052a10cd7d401a5c04f83f371a06fe293c
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.3
Operating System: linux
GOPATH=/home/rakan/Code/Go
GOROOT=/usr/lib/go
```

**Comment by LefterisJP** [comment]
Okay @rakanalh can you add all these info in the first comment of the ticket then? That means then that a similar issue exists for geth.

**Comment by rakanalh** [comment]
Updated the issue description with additional data as required.

**Comment by andrevmatos** [comment]
Duplicate of #1785

**Comment by andrevmatos** [comment]
Fixed by #1786
